importers/tuition-fees-importer.php

```diff
@@ -233,15 +233,24 @@ private function update_degrees() {
 			$parent_program_type    = wp_get_post_terms( $degree->ID, 'program_types', array( 'parent' => 0 ) );
 			$parent_program_type_id = is_array( $parent_program_type ) ? $parent_program_type[0]->term_id : 0;
 			$program_type = wp_get_post_terms( $degree->ID, 'program_types', array( 'parent' => $parent_program_type_id ) );
-			$program_type = ( is_array( $program_type ) && ! empty( $program_type ) ) ? $program_type[0] : null;
+			if ( is_array( $program_type ) && ! empty( $program_type ) ) {
+				$program_type = $program_type[0];
+			}
+			elseif ( $parent_program_type_id > 0 ) {
+				$program_type = $parent_program_type[0];
+			}
+			else {
+				$program_type = null;
+			}
+
 			$plan_code    = get_post_meta( $degree->ID, UCF_Tuition_Fees_Config::get_option_or_default( 'degree_plan_code_name' ), true );
 			$subplan_code = get_post_meta( $degree->ID, UCF_Tuition_Fees_Config::get_option_or_default( 'degree_subplan_code_name' ), true );
 			$is_online    = filter_var( get_post_meta( $degree->ID, UCF_Tuition_Fees_Config::get_option_or_default( 'degree_online_meta_field' ), true ), FILTER_VALIDATE_BOOLEAN );
 
 			// If no program type, skip it
 			if ( ! $program_type ) { $this->skipped_total++; continue; }
 
-			$schedule_code = $this->get_schedule_code( $program_type->name, $plan_code, $subplan_code, $is_online );
+			$schedule_code = $this->get_schedule_code( $degree, $program_type->name, $plan_code, $subplan_code, $is_online );
```

Comment:
The `$program_type->name` is going to throw a warning if `$program_type` is null (which is an option based on line 243). We can avoid the warning in the log by putting the name in a variable, or setting it to null if `$program_type` is null.